### DISCUSSION TITLE
Cannot stop fs event from start fs event callback ?

### DISCUSSION THREAD

**Discussion by 5415nofar1** [libuv/libuv#3255] [Q&A] [answered]
I am listening to changes on a particular file. When the file changes I stop listening and do some things.

It runs and works fine...

But in case of debugging in Windows I get an assertion here:
`static void uv_fs_event_queue_readdirchanges(uv_loop_t* loop,
    uv_fs_event_t* handle) {
  assert(handle->dir_handle != INVALID_HANDLE_VALUE);`

My code looks like this:

`uv_fs_event_start(&fs_event_req, OnChangeCb, file_path, 0);`

`void OnChangeCb(uv_fs_event_t* req, const char* filename, int events, int status){
   ....
   uv_fs_event_stop(&fs_event_req);
   .... do some things
}`

I noticed that the problem is related to the fact that at the beginning of the function `uv_process_fs_event_req` the request handle is active and after the callback `OnChangeCb` which call to `uv_fs_event_stop` function, the request is no longer active and despite this the function `uv_fs_event_queue_readdirchanges` is called and the assert goes up...

Is this a bug? Did I use the functions incorrectly?
Thank you!
Nofar

**Comment by vtjnash** [comment, marked as answer]
Sounds like a bug, can you make an issue (or PR 😉)? Many users do `uv_fs_event_start` / `uv_close` for managing object lifetime (as do all of the libuv tests), so this seems to have just been overlooked.
